MhLabs.AwsSignedHttpClient/Extensions/ServiceCollectionExtension.cs

```diff
@@ -28,26 +30,27 @@ public static class ServiceCollectionExtension
         private static IServiceCollection AddMhHttpClient<TClient, TImplementation>(this IServiceCollection services, HttpOptions options = null) where TClient : class
             where TImplementation : class, TClient
         {
+            
             if (options == null) options = new HttpOptions();
 
             var httpClientBuilder = services.AddHttpClient<TClient, TImplementation>(client =>
-                {
-                    client.BaseAddress = GetBaseUrl(options);
-                }).AddHttpMessageHandler<AwsSignedHttpMessageHandler>()
+            {
+                client.BaseAddress = GetBaseUrl(options);
+            }).AddHttpMessageHandler<AwsSignedHttpMessageHandler<TClient>>()
```

Comment:
This should be typed, i.e. AddUnsignedHttpClient should not use AwsSginedHttpMessageHandler